Gems/ROS2/Code/Source/Clock/SimulationClock.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "SimulationClock.h"
+#include <ROS2/ROS2Bus.h>
+#include <rclcpp/qos.hpp>
+
+namespace ROS2
+{
+    builtin_interfaces::msg::Time SimulationClock::GetROSTimestamp() const
+    {
+        auto elapsedTime = GetElapsedTimeMicroseconds();
+
+        builtin_interfaces::msg::Time timeStamp;
+        timeStamp.sec = static_cast<int32_t>(elapsedTime / 1000000);
+        timeStamp.nanosec = static_cast<uint32_t>((elapsedTime % 1000000) * 1000);
+        return timeStamp;
+    }
+
+    int64_t SimulationClock::GetElapsedTimeMicroseconds() const
+    {
+        if (auto* timeSystem = AZ::Interface<AZ::ITime>::Get())
+        {
+            return static_cast<int64_t>(timeSystem->GetElapsedTimeUs());
+        }
+        else
+        {
+            AZ_Warning("ROS2SystemComponent", false, "No ITime interface available");
```

Comment:
Clarification: Is this an error or a warning? If the elapsed time is critical and is used by other parts of the system, this should be an error, unless downstream subscribers know how to handle 'zero' elapsed times

Reply:
This would be an error for the simulation, since ros2 packages would not be able to run many time-sensitive algorithms (e.g. sensor fusion). I escalated it in the new changeset.